sdk/demos/intel/fluid/demo.js

```diff
@@ -0,0 +1,693 @@
+    -gMapWidth/2.0, -gMapHeight / 2.0, 0, 1,
+  ];
+  let mProjection = orthographic(-gMapWidth/2.0, gMapWidth/2.0, -gMapHeight / 2.0,
+      gMapHeight / 2.0, 0, 1);
+  gViewProjection.loadIdentity();
+  gViewProjection.multiply(mView);
+  gViewProjection.multiply(mProjection);
+
+  gl.bindBufferBase(gl.SHADER_STORAGE_BUFFER, 0, gParticleBuffers[0]);
+  gl.bindBufferBase(gl.SHADER_STORAGE_BUFFER, 1, gParticleBuffers[1]);
+  gl.bindBufferBase(gl.SHADER_STORAGE_BUFFER, 2, gParticleBuffers[2]);
+  gl.bindBufferBase(gl.SHADER_STORAGE_BUFFER, 3, gParticleBuffers[3]);
+  gl.bindBufferBase(gl.SHADER_STORAGE_BUFFER, 4, gParticleBuffers[4]);
+  gl.bindBufferBase(gl.SHADER_STORAGE_BUFFER, 5, gParticleBuffers[5]);
+  gl.bindBufferBase(gl.SHADER_STORAGE_BUFFER, 6, gParticleBuffers[6]);
+}
+
+function initRender() {
+  gCanvas.width = window.innerWidth;
+  gCanvas.height = window.innerHeight;
+  gl.viewport(0, 0, gCanvas.width, gCanvas.height);
+  gl.clearColor(0.0, 0.0, 0.0, 1.0);
+
+  gPointSize = 4.0 * gCanvas.width * gCanvas.height / (1024 * 1024);
+
+  let vertexShader = loadShader(gl.VERTEX_SHADER, gShaderStr[VS]);
+  let fragmentShader = loadShader(gl.FRAGMENT_SHADER, gShaderStr[FS]);
+  gPrograms[VS] = gl.createProgram();
+  gl.attachShader(gPrograms[VS], vertexShader);
+  gl.attachShader(gPrograms[VS], fragmentShader);
+
+  // Bind a_particlePos to attribute 0
+  // Bind a_particleVel to attribute 1
+  gl.bindAttribLocation(gPrograms[VS], 0, 'a_particlePos');
+  gl.bindAttribLocation(gPrograms[VS], 1, 'a_particleDen');
+
+  // Load the vertex data
+  gl.bindBuffer(gl.ARRAY_BUFFER, gParticleBuffers[0]);
+  gl.enableVertexAttribArray(0);
+  gl.vertexAttribPointer(0, 2, gl.FLOAT, gl.FALSE, 16, 0);
+  gl.vertexAttribDivisor(0, 1);
+
+  gl.bindBuffer(gl.ARRAY_BUFFER, gParticleBuffers[1]);
+  gl.enableVertexAttribArray(1);
+  gl.vertexAttribPointer(1, 1, gl.FLOAT, gl.FALSE, 16, 0);
+  gl.vertexAttribDivisor(1, 1);
+
+  // Link the program
+  gl.linkProgram(gPrograms[VS]);
+  gl.useProgram(gPrograms[VS]);
+
+  // Set uniforms
+  let mViewProjectionLoc = gl.getUniformLocation(gPrograms[VS], 'u_viewProjection');
+  gl.uniformMatrix4fv(mViewProjectionLoc, false, gViewProjection.elements);
+  let fPointSizeLoc = gl.getUniformLocation(gPrograms[VS], 'u_fPointSize');
+  gl.uniform1f(fPointSizeLoc, gPointSize);
+}
+
+function initCS() {
+  [CS_DEN_SIM, CS_DEN_SHA, CS_FOR_SIM, CS_FOR_SHA, CS_INT, CS_BUILD_GRID,
+    CS_BITONIC_SORT, CS_TRANSPOSE, CS_CLEAR_GRID_IND, CS_BUILD_GRID_IND,
+    CS_REARRANGE, CS_DEN_GRID, CS_FOR_GRID, CS_INT_GRID]
+  .forEach((v) => {
+    let cs = loadShader(gl.COMPUTE_SHADER, gShaderStr[v]);
+    gPrograms[v] = gl.createProgram();
+    gl.attachShader(gPrograms[v], cs);
+    gl.linkProgram(gPrograms[v]);
+    gl.useProgram(gPrograms[v]);
+  });
+}
+
+function destroyBuffers() {
+  gl.deleteBuffer(gParticleBuffers[0]);
+  gl.deleteBuffer(gParticleBuffers[1]);
+  gl.deleteBuffer(gParticleBuffers[2]);
+  gl.deleteBuffer(gUpdateParams);
```

Comment:
It seems that you need to delete more buffers range from 0~6. And please don't forget to delete all uniform buffers.

Reply:
Done.